### DISCUSSION TITLE
GPT (and other LLMs) with Superset: Use cases and experimentation

### DISCUSSION THREAD

**Discussion by rusackas** [apache/superset#24178] [General]
As discussed in a recent blog post, and the corresponding meetup, there are NUMEROUS ways that the new wave of Large Language Models (LLMs, such as GPT, Bard, and others) are able to provide new opportunities for the Superset user. These may take the shape of new workflows, integrations, or features. We'd love to hear your feedback on the things we've tried so far, things you think are worth exploring more deeply, and other ideas we might not yet have thought of. Hopefully we can turn this thread into future experiments/content, and product features if all goes well! We welcome your input, and we encourage your upvoting of ideas/points in this thread that you want to support!

(Thread will remain locked until the meetup ends)

**Comment by rusackas** [comment]
From a meetup suggestion: Give [Perplexity](https://www.perplexity.ai/) a whirl in some of these tests.

**Comment by rusackas** [comment]
Some of these tests (particularly the dev ones) should probably also be tested with Github CoPilot (or similar IDE integrations)

**Comment by JoeHowie** [comment]
I gather from the presentation (good job!) that LLM performance drafting sql queries is greatly improved by having a textual explanation of the reason and purpose behind the tables and columns of interest and an explanation of how/why they're linked. Should there be a standard to create and store those narratives in the postgres comments section for each table and perhaps even at the column level and then have Superscript read and use those comments? This could help result in more consistent and better results and lessen the demands on users to create or know how to use such descriptions.

**Reply by JoeHowie** [reply]
Something like the following query could be used to pull up the database comments/descriptions:
SELECT RelName,Description 
FROM pg_Description
JOIN pg_Class 
ON pg_Description.ObjOID = pg_Class.OID
WHERE ObjSubID = 0

**Comment by chiweng2009** [comment]
I mentioned in some WeChat group that I hope to use LLM to understand users' naturally expressed intentions, and quickly render charts with superset.
My idea is that nl2sql recognizes natural language, converts it into SQL, and then superset can provide lightweight and simplified APIs

**Reply by ved-kashyap-samsung** [reply]
My team has been working on this similar feature based on our requirement. Can you please let us know if we can raise PR for this feature under some feature flag. @rusackas

**Reply by rusackas** [reply]
While I'm VERY curious how you're approaching this (and of course you're welcome to open a PR!) this seems like the kind of integration that will need to undergo some scrutiny to make sure that it's an LLM integration that the community will want to take under its wing - from a security standpoint, a maintenance footprint, and of course the first AI feature will likely lead to many more, so there'll be some need to assess the architectural implications. a [SIP](https://github.com/apache/superset/issues/5602) might be warranted, or if the PR is easier, then we can certainly take a look and raise discussions around it as warranted. Thanks either way!

**Reply by ketulgupta1995** [reply]
@ved-kashyap-samsung , would be interesting to go through it , is a PR already created or SIP ?
